Source/HTTPHeaders.swift

```diff
@@ -24,40 +24,301 @@
 
 import Foundation
 
-public typealias HTTPHeaders = [String: String]
 
-extension Dictionary where Key == String, Value == String {
-    public static func authorization(username: String, password: String) -> HTTPHeaders {
-        let credential = Data("\(username):\(password)".utf8).base64EncodedString()
+/// An order-preserving and case-insensitive representation of HTTP headers.
+public struct HTTPHeaders {
+    private var headers = [HTTPHeader]()
+
+    /// Create an empty instance.
+    public init() { }
+
+    /// Create an instance from an array of `HTTPHeader`s. Duplicate case-insensitive names are collapsed into the last
+    /// name and value encountered.
+    public init(_ headers: [HTTPHeader]) {
+        self.init()
+
+        headers.forEach { update($0) }
+    }
+
+    /// Create an instance from a `[String: String]`. Duplicate case-insensitive names are collapsed into the last name
+    /// and value encountered.
+    public init(_ dictionary: [String: String]) {
+        self.init()
+
+        dictionary.forEach { update(HTTPHeader(name: $0.key, value: $0.value)) }
+    }
+
+    /// Case-insensitively updates or appends an `HTTPHeader` into the instance using the provided `name` and `value`.
+    ///
+    /// - Parameters:
+    ///   - name:  The `HTTPHeader` name.
+    ///   - value: The `HTTPHeader value.
+    public mutating func update(name: String, value: String) {
+        update(HTTPHeader(name: name, value: value))
+    }
+
+    /// Case-insensitively updates or apprends the provided `HTTPHeader` into the instance.
+    ///
+    /// - Parameter header: The `HTTPHeader` to update or append.
+    public mutating func update(_ header: HTTPHeader) {
+        guard let index = headers.index(of: header.name) else {
+            headers.append(header)
+            return
+        }
+
+        headers.replaceSubrange(index...index, with: [header])
+    }
 
-        return ["Authorization": "Basic \(credential)"]
+    /// Case-insensitively removes an `HTTPHeader`, if it exists, from the instance.
+    ///
+    /// - Parameter name: The name of the `HTTPHeader` to remove.
+    public mutating func remove(name: String) {
+        guard let index = headers.index(of: name) else { return }
+
+        headers.remove(at: index)
+    }
+
+    /// Sort the current instance by header name.
+    mutating public func sort() {
+        return headers.sort { $0.name < $1.name }
```

Comment:
I think you could ditch the `return` here.

Reply:
Fixed in [aaa3e6a](
https://github.com/Alamofire/Alamofire/pull/2629/commits/aaa3e6a10d2b14ad0dbc049dc5aa3b1b9bb8fa2e).